modules/oe_theme_content_event/src/Plugin/ExtraField/Display/StatusMessage.php

```diff
@@ -0,0 +1,202 @@
+        $this->applyHourTag($build, $event->getOnlineStartDate());
+        $livestream_date = $this->dateFormatter->format($entity->get('oe_event_online_dates')->start_date->getTimestamp(), 'oe_event_date_hour_timezone');
+        $build['#title'] = $this->t('This event has started. The livestream will start at @date.', ['@date' => $livestream_date]);
+        return $build;
+      }
+
+      // If the livestream is ongoing, cache it by its end date.
+      if ($event->isOnlinePeriodActive($this->requestDateTime)) {
+        $this->applyHourTag($build, $event->getOnlineEndDate());
+
+        // The event is over, apply timezone cache contexts.
+        if ($event->isOver($this->requestDateTime)) {
+          $build['#title'] = $this->t('This event has ended, but the livestream is ongoing.');
+          $cacheable->addCacheContexts(['timezone']);
+          $cacheable->applyTo($build);
+          return $build;
+        }
+
+        // If event is not started, cache it by its start date.
+        if ($this->requestDateTime < $event->getStartDate()->getPhpDateTime()) {
+          $build['#title'] = $this->t('The livestream has started.');
+          $this->applyHourTag($build, $event->getStartDate());
+          return $build;
+        }
+
+        // If event is ongoing, cache it by its end date.
+        if ($event->isOngoing($this->requestDateTime)) {
+          $build['#title'] = $this->t('This event has started. You can also watch it via livestream.');
+          $this->applyHourTag($build, $event->getEndDate());
+          return $build;
+        }
+      }
+    }
+
+    // If the event status is 'As planned', we'll use the 'info' variant and
+    // add the title based on the event state (ongoing or past).
+    if ($event->isAsPlanned()) {
+      $build['#variant'] = 'info';
+      $build['#icon'] = 'information';
+      if ($event->isOver($this->requestDateTime)) {
+        // Add timezone cache contexts.
+        $cacheable->addCacheContexts(['timezone']);
+        $cacheable->applyTo($build);
+        $build['#title'] = $this->t('This event has ended.');
+        return $build;
+      }
+      // If event is ongoing, cache it by its end date.
+      if ($event->isOngoing($this->requestDateTime)) {
+        $build['#title'] = $this->t('This event has started.');
+        $this->applyHourTag($build, $event->getEndDate());
+        return $build;
+      }
+    }
+
+    // If we reached this point, then the message will use the 'warning' variant
+    // and the title will reflect the event status.
+    $build['#variant'] = 'warning';
+    $build['#icon'] = 'warning';
+    if ($event->isPostponed()) {
+      $build['#title'] = $this->t('This event has been postponed.');
+    }
+    if ($event->isCancelled()) {
+      $build['#title'] = $this->t('This event has been cancelled.');
+    }
+    if ($event->isRescheduled()) {
+      $build['#title'] = $this->t('This event has been rescheduled.');
+    }
```

Comment:
These will be never reached if we have online type and dates set even though the event status is set to one of these.

Reply:
The livestream messages have priority over these ones, as requested.

Regarding online type/dates/link, according to Evegenii's [comment](https://github.com/openeuropa/oe_theme/pull/997#discussion_r764186721) to avoid inconsistency, I'll check the Online type instead of the Online link as requested in the ticket.

Reply:
Fixed the logic displaying event status messages (if it's not as planned) over livestream ones.